core/offchain/impl/uri.hpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_OFFCHAIN_URI
+#define KAGOME_OFFCHAIN_URI
+
+#include <string>
+#include <string_view>
+
+namespace kagome::offchain {
+  struct Uri final {
+   public:
+    std::string Schema;
+    std::string Host;
+    std::string Port;
+    std::string Path;
+    std::string Query;
+    std::string Fragment;
```

Comment:
This might be economy on matches but it could make sense to store the Uri as a string and store these fields as string_views into that string.

Reply:
Why not? Refactored.